### ISSUE TITLE
10.8.1 SDL3 Playfield renders black

### ISSUE THREAD

**Issue by Meshuggah70** [vpinball/vpinball#1956]
**Describe the bug**
In the new SDL3 versions  of GL 10.81 Vpx is Looking for SDL3.dll and the DLL file is named  SDL364.dll had to rename to get it to open.. PF is black though on all versions.. maybe missing a file as well or path

**Comment by Meshuggah70** [comment]
Fixed in update 2647... screen still loads black and cant use though

**Comment by toxieainc** [comment]
Could you please write more about your config (GPU, OS, etc)?

**Comment by Meshuggah70** [comment]
I'm running a Ryzen 9 5050X,  RTX4070ti with windows 11.. It almost looks
like it does this switch that you see when an Nvidia changes modes but the
playfield screen is blackn. I see the back glass and the DMD On some tables
I can hear when I insert the coin but you can't see anything and you can't
start the game. I'm pretty sure the issue has to do with SDL3.. because any
version before that works fine any version after that I get a black screen.
I'm running yesterday's MAME 3.6 and B2S server version

**Comment by jsm174** [comment]
fwiw, SDL3 is still in preview, as well as 10.8.1. Unless your testing, I thought the recommendation was to stick with 10.8.0.

Also, nvidia just put out a brand new driver on 10/22.

**Comment by Meshuggah70** [comment]
Yes I have the brand new driver and yes I am testing. I'm not an "Official"
tester But I figured if I was developing something I would like to know
when something wasn't working correctly so I could work on the problem..

**Comment by jsm174** [comment]
> Yes I have the brand new driver and yes I am testing. I'm not an "Official" tester But I figured if I was developing something I would like to know when something wasn't working correctly so I could work on the problem.. I would also assume this section is for reporting bugs which I did..Sorry was just trying to help out

I think you took my reply the wrong way. 😄  We definitely appreciate the feedback and testing.

**Comment by Meshuggah70** [comment]
No problem. I enjoy testing each version .BTW was really impressed with RC5
Looks like it's shaping up very well.

**Comment by toxieainc** [comment]
Do you please have a screenshot how exactly it looks?

**Comment by Meshuggah70** [comment]
Yes I can when I get home later on tonight but the Playfield is completely blank (black screen) I can see the back glass and see the DMD underlay and can hear whenever I put coins in but nothing else ( on some tables) others nothing at all

**Comment by Meshuggah70** [comment]
> Do you please have a screenshot how exactly it looks?

Here is a video

https://drive.google.com/file/d/1YDk7L4IbWu9V2zN5SQBDXQkQWJXm8ykw/view?usp=drivesdk

**Comment by pacleary** [comment]
Testing 10.8.1 as well. 

In the 64 OpenGL Debug version I get:

"Assertion Failed!"

Program: C:vPinball\VisualPinballBeta\VPinallX_Gl64.exe
File: D:\a\a\vpinball\vpinball\src\renderer\Shader.h
Line: 576

Expression: m_shader->m_stateOffsets[uniformName] != -1


In the non-Debug, it opens with the same black screen.

I have this installed in a separate folder from VPX 10.8 RC5, so I suspect that might be the issue?

**Comment by Crrispy** [comment]
Hello,

I have the same issue here after upgrading to 10.8.1 2675, Windows 11. Black playfield with only a few traces of something appearing. Reverting to 2667 for now.

**Comment by francisdb** [comment]
ping @vbousquet 

I was trying Indiana Jones VPW today and got this in the logs:

```
2024-11-08 08:55:37.618 INFO  [7885] [Shader::Load@2137] Parsing file FBShader.glfx
2024-11-08 08:55:37.696 ERROR [7885] [Shader::getUniformByName@434] [FBShader.glfx] getUniformByName Could not find uniform exposure in shaderUniformNames
2024-11-08 08:55:37.721 ERROR [7885] [Shader::getUniformByName@434] [FBShader.glfx] getUniformByName Could not find uniform exposure in shaderUniformNames
...
```

10.8.1 GL branch from today renders black, see screenshots below
10.8.1 BGFX build works mostly but logs this

```
2024-11-08 09:04:44.364 INFO  [16585] [RenderDevice::RenderDevice@532] BGFX initialized using Vulkan backend
2024-11-08 09:04:44.365 ERROR [16585] [Shader::getUniformByName@434] [] getUniformByName Could not find uniform exposure in shaderUniformNames
2024-11-08 09:04:44.365 ERROR [16585] [Shader::loadProgram@1350] Invalid uniform defined in shader fs_dmd2_srgb_noclip: exposure
```

![image](https://github.com/user-attachments/assets/e9012a45-d54c-42b1-a9fe-cb3a7fcbf513)

Rendering the tweak menu works

![image](https://github.com/user-attachments/assets/2ded7c77-7a2c-402d-9a80-31ebbbcf5293)

**Comment by francisdb** [comment]
@toxieainc @toxieainc can somebody please fix that broken shader?

**Comment by francisdb** [comment]
Is there a way to have the shaders compiled on CI to catch these issues?

**Comment by jsm174** [comment]
GL shaders are compiled during runtime, so I don’t believe so.

**Comment by francisdb** [comment]
Tests are "runtime". Also steam ships precompiled shaders when updating games so there must be something that can be done?

**Comment by francisdb** [comment]
But the actual error for this case might indeed be a runtime error. Anyway. I guess @vbousquet knows what is going on.

**Comment by Crrispy** [comment]
Hello,

Still the same in 2703 DX. Playfield is solid black.

**Comment by francisdb** [comment]
Seems to be fixed for me on the latest build of 10.8.1? Is it for you @Meshuggah70 ?

**Comment by Meshuggah70** [comment]
Will give it a try tonight

**Comment by pacleary** [comment]
10.8.1 Beta 2710 OpenGL is rendering properly now. No more black screen.

Although now the table options crash and close the program. Is that happening for anyone else? Not sure how often I should be making issue posts given how frequently this is being updated.

**Comment by francisdb** [comment]
As a dev I personally like to know about issues asap as then I still have the context around. It also helps other users that might encounter the same bug.
But not sure if vpinball main devs like issues for 10.8.1 as they for example did not close this one when they fixed the code.

**Comment by jsm174** [comment]
I just tested the latest on iOS (since it does some LiveUI hacking), and it crashes as well, so has to be something in the last few commits.

**Comment by francisdb** [comment]
Thanks @vbousquet

**Comment by toxieainc** [comment]
Reopening as @Meshuggah70 says this is still happening for him under GL.

**Comment by vbousquet** [comment]
@Meshuggah70 Could you have a try with latest build ? I think this should be solved now

**Comment by vbousquet** [comment]
I will close as we had 5+ month without any news and it is working on my end
